src/backend/dx11/src/device.rs

```diff
@@ -1391,6 +1492,10 @@ impl hal::Device<Backend> for Device {
             let ptr = host_visible.borrow_mut().as_mut_ptr();
             memory.mapped_ptr.replace(Some(ptr));
 
+            if memory.ty == MemoryHeapType::HOST_COHERENT {
+                memory.invalidate(&self.context, memory.resolve(&(None, None)));
```

Comment:
This is not actually correct. AFAIK, the operation of mapping/unmapping memory doesn't involve any synchronization in Vulkan *regardless* of the memory coherency:
  - non-coherent memory has to still be explicitly sync-ed with flush/invalidate
  - coherent memory still has to affect all subsequent device and queue based commands (e.g. submissions), even if not unmapped.

Reply:
Ouch!

Do we need to put in checks on submissions etc. and flush mapped coherent memory then? Sounds a bit messy but not impossible..

Reply:
Well, putting the sync in map/unmap is simply incorrect. It's not even a subset of the proper logic, as I explained. And if you just remove it, then coherent memory wouldn't be synced at all... I know, this sucks. And ultimately the main client of your backend is WebRender, which can just use non-coherent memory. So if you can think of a quick and easy way to provide at least some coherency guarantees (that are a subset of the end goal), go for it. Otherwise, let's try implementing the proper logic in place.

Reply:
Not trivially. The copies to/from are command buffer operations. It's not correct to synchronize at command buffer *recording*, you'll have to defer this to submission time.